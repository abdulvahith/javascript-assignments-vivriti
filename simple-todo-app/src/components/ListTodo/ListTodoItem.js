import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Container from "@material-ui/core/Container";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import Divider from "@material-ui/core/Divider";
import Checkbox from "@material-ui/core/Checkbox";
import Button from "@material-ui/core/Button";
import Styles from "./ListTodo.css.js";

export default class ListTodoItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isEdit: false,
      todoEditVal: props.todoData
    };
    this.showEdit = this.showEdit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.editTodo = this.editTodo.bind(this);
    this.deleteTodo = this.deleteTodo.bind(this);
    this.markAsComplete = this.markAsComplete.bind(this);
  }

  onChange(e){
    this.setState({
      todoEditVal: e.target.value
    })
  }

  showEdit() {
    this.setState({
      isEdit: true
    });
  }

  editTodo(){
    let { editTodoItem, todoId }= this.props;
    this.setState({
      isEdit: false
    });
    editTodoItem({ id: todoId, todoVal: this.state.todoEditVal })
  }

  deleteTodo(){
    let { deleteTodoItem, todoId }= this.props;
    deleteTodoItem(todoId);
  }

  markAsComplete(){
    let { completeTodoItem, todoId, isFromComplete }= this.props;
    completeTodoItem(todoId, !isFromComplete);
  }

  render() {
    let { todoData, isFromComplete } = this.props;
    let { todoEditVal }= this.state;

    return (
      <ListItem>
        <Grid container spacing={4}>
          <Grid item xs={1}>
            <Checkbox
              checked={isFromComplete}
              onClick={this.markAsComplete}
              color="primary"
              id='list-checkBox'
            />
          </Grid>
          {this.state.isEdit ? (
            <React.Fragment>
              <Grid item xs={5}>
                <input value={todoEditVal} onChange={this.onChange} id='list-checkbox'/>
              </Grid>
              <Grid item xs={3}>
                <Button
                  size="medium"
                  style={Styles.margin_common}
                  onClick={this.editTodo}
                  id='list-save-btn'
                >
                  Save
                </Button>
              </Grid>
            </React.Fragment>
          ) : (
            <React.Fragment>
              <Grid item xs={5}>
                <Typography
                  style={isFromComplete ? Styles.compelted_text : Styles.margin_common }
                  variant="subtitle1"
                  display="block"
                  gutterBottom
                >
                  {todoData}
                </Typography>
              </Grid>
              <Grid item xs={3}>
                <Button
                  size="medium"
                  id='list-edit-btn'
                  style={Styles.margin_common}
                  onClick={this.showEdit}
                >
                  Edit
                </Button>
              </Grid>
            </React.Fragment>
          )}
          <Grid item xs={3}>
            <Button size="medium" style={Styles.margin_common} onClick={this.deleteTodo} id='list-delete-btn'>
              Delete
            </Button>
          </Grid>
        </Grid>
      </ListItem>
    );
  }
}

ListTodoItem.propTypes = {
  todoData: PropTypes.string,
  todoId: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  editTodoItem: PropTypes.func,
  deleteTodoItem: PropTypes.func,
  completeTodoItem: PropTypes.func,
  isFromComplete: PropTypes.bool
}
