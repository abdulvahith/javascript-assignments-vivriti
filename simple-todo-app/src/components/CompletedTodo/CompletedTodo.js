import React, { Component } from "react";
import PropTypes from "prop-types";
import {connect} from 'react-redux';
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import Divider from "@material-ui/core/Divider";
import Checkbox from "@material-ui/core/Checkbox";
import Button from "@material-ui/core/Button";
import Styles from "./CompletedTodo.css.js";
import List from "@material-ui/core/List";
import ListTodoItem from '../ListTodo/ListTodoItem'

export default class CompletedTodo extends Component {
  render() {
    let { todoList, editTodoItem, deleteTodoItem, completeTodoItem } = this.props;

    let completedItem = todoList
      ? todoList.map((item, indes) => {
          if (item.isCompleted) {
            return (
              <ListTodoItem
                todoData={item.data}
                key={item.id}
                todoId={item.id}
                editTodoItem= {editTodoItem}
                deleteTodoItem= {deleteTodoItem}
                completeTodoItem= {completeTodoItem}
                isFromComplete= {true}
              />
            );
          }
        })
      : null;

    return (
      <Container maxWidth="sm" style={Styles.list_container}>
        <Typography variant="h5" component="h2">
          Completed List
        </Typography>
        <Divider style={Styles.list_divider} />
        <List>{completedItem}</List>
      </Container>
    );
  }
}
