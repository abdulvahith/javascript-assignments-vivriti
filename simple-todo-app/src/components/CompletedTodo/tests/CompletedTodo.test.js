import React from 'react';
import renderer from 'react-test-renderer';
import CompletedTodo from '../CompletedTodo';
import ListTodoItem from '../../ListTodo/ListTodoItem';
import { mount, shallow } from 'enzyme';

describe('CompletedTodo component test add and input value', () => {
  it('renders without crashing', () => {
    shallow(<CompletedTodo />);
  });

  it('renders correctly snapshot test', () => {
   // const todo = { id: 1, done: false, name: 'Buy Milk' };
   const rendered = renderer.create(
     <ListTodoItem
       todoData="test"
       key="1234"
       todoId="1234"
       isFromComplete= {true}
     />
   );
   expect(rendered.toJSON()).toMatchSnapshot();
 });

  it('call delete, edit and complete function', () => {
    const editTodoItem = jest.fn();
    const deleteTodoItem = jest.fn();
    const completeTodoItem = jest.fn();

    const wrapper = shallow(
      <ListTodoItem
        todoData="test"
        key="1234"
        todoId="1234"
        editTodoItem= {editTodoItem}
        deleteTodoItem= {deleteTodoItem}
        completeTodoItem= {completeTodoItem}
        isFromComplete= {true}
      />
    );

    const editBtn = wrapper.find('#list-edit-btn');
    editBtn.simulate('click');
    setTimeout(() => {
      //after clicks edit button should trigger save button
        const saveBtn = wrapper.find('#list-save-btn');
        saveBtn.simulate('click');
        expect(editTodoItem).toHaveBeenCalled();
      }, 500);
    const deleteBtn = wrapper.find('#list-delete-btn');
    const listCheckBox = wrapper.find('#list-checkBox');
    deleteBtn.simulate('click');
    expect(deleteTodoItem).toHaveBeenCalled();
    listCheckBox.simulate('click');
    expect(completeTodoItem).toHaveBeenCalled();
  });

});
