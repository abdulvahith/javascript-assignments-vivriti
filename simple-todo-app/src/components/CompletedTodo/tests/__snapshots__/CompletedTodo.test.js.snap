// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedTodo component test add and input value renders correctly snapshot test 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-121 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-122 Mui-checked MuiIconButton-colorPrimary"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-124"
            data-indeterminate={false}
            id="list-checkBox"
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom MuiTypography-displayBlock"
        style={
          Object {
            "marginTop": "7px",
            "textDecoration": "line-through",
          }
        }
      >
        test
      </h6>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        id="list-edit-btn"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginTop": "7px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Edit
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        id="list-delete-btn"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginTop": "7px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</li>
`;
